src/activities/mirrorgame/mirrorgame.js

```diff
@@ -0,0 +1,536 @@
+             [0,0,0,0,0]
+              ]
+            ,
+            [//level2
+             [0,0,0,0,0],
+             [0,1,0,0,0],
+             [0,0,2,0,0],
+             [0,2,0,0,0],
+             [0,0,3,0,0]
+            ],
+            [//level3
+             [1,0,3,0,0],
+             [0,0,0,0,0],
+             [0,0,0,0,2],
+             [0,0,0,2,0],
+             [0,0,0,0,0]
+            ],
+            [//level4
+             [0,0,1,0,0],
+             [0,0,0,2,0],
+             [0,0,2,0,0],
+             [0,0,0,2,0],
+             [0,0,3,0,0]
+            ],
+            [//level5
+             [0,0,0,0,0],
+             [1,0,0,0,3],
+             [0,2,0,0,0],
+             [2,0,0,0,0],
+             [0,2,0,0,0]
+            ]
+                    ]
+var currentLevel = 0
+var numberOfLevel = 5
+var arr
+var items
+var count = 0
+var num = 0
+var numberofmirrors = 0
+var i,j
+var Light
+var currLight
+var mirror
+var torch
+var bulbon
+var bulboff
+
+
+
+
+function start(items_) {
+    items = items_
+    currentLevel = 0
+     Light = new Array
+    items.ok.visible = false;
+
+
+    createLightgrid()
+    initLevel()
+
+}
+
+function stop() {
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+            arr = data[currentLevel]
+    currLight = new Array
+    for(i=0;i<items.rows;i++)
+        currLight[i] = new Array
+
+    for(i=0;i<items.rows;i++)
+        for(j=0;j<items.columns;j++)
+        currLight[i][j] = new Array
+
+    mirror = new Array
+    for(i=0;i<items.rows;i++)
+        mirror[i] = new Array
+
+
+    bulbon = new Array
+    bulboff = new Array
+    items.ok.visible = false;
+    for(var y = 0;  y < items.rows; y++) {
+        for(var x = 0;  x < items.columns; x++) {
+
+            items.pieces.append({'stateTemp': "invisible"})
+            if(arr[y][x]===1)
+            {createTorch(x,y);
+                for(var k = 0;k < count;k++) {
+                    for(var row = 0; row < 5;row++) {
+                        for(var column = 0; column < 5; column++) {
+                    if(Light[k].uniquename === "lightpath"+y+x+row+column)
+                        Light[k].visible =true;
+                        }
+                     }
+                }
+            }
+            if(arr[y][x]===2)
+            {createMirror(x,y);numberofmirrors++;}
+            if(arr[y][x]===3)
+            {createBulb(x,y);}
+
+            }
+
+        }
+    checkLightPath();
+    }
+
+function mod(a,b)
+{   return (a>=b)? (a-b):(b-a);}
+
+function min(a,b)
+{return (a<=b)?a:b;}
+
+function max(a,b)
+{return (a<=b)?b:a;}
+
+function createLightgrid()
+{
+    for(var x = 0;x < items.rows; x++) {
+         for(var y = 0;  y < items.columns; y++){
+             for(var w = x+1; w < items.rows; w++){
+                 for(var z = y+1; z < items.columns; z++){
+                     if((x-w) === (y-z) && x!==w && y!==z)
+                     {
+                 var component = Qt.createComponent("Light-45.qml");
```

Comment:
i have created the light grid once only...for the grid part i have to add mirrors simultaneously with grid ,that's why i am creating it at every level